tools/prologs-epilogs/sudo-oslogin

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# On Google Cloud VMs that have OS Login enabled, this script will query the
+# OS Login API and add users as sudoers if they have the OS Admin Login role.
+# The sudoers entry is added at job start and removed upon job completion.
+
+# ensure that older Slurm NSS configurations are removed
+sed -i 's,slurm ,,' /etc/nsswitch.conf
+
+# only run this script if OS Login is enabled
+i_config=$(curl -s -H "Metadata-Flavor: Google" \
```

Comment:
Since `jq` is extremely useful later in the script and `jq` is installed in all Slurm images, I've chosen to leave it here, but add an early exit if `jq` is not in the PATH.